src/blueprint/from_config.rs

```diff
@@ -29,11 +29,14 @@ type Valid<A> = ValidDefault<A, String>;
 pub fn config_blueprint(config: &Config) -> Valid<Blueprint> {
   let output_types = config.output_types();
   let input_types = config.input_types();
-  let schema = to_schema(config)?;
-  let definitions = to_definitions(config, output_types, input_types)?;
-  let server: Server = Server::try_from(config.server.clone())?;
+  let schema = to_schema(config);
+  let definitions = to_definitions(config, output_types, input_types);
+  let server: Valid<Server> = Server::try_from(config.server.clone());
   let upstream = config.upstream.clone();
-  valid_base_url(upstream.base_url.as_ref())?;
+  let (((_, schema), definitions), server) = valid_base_url(upstream.base_url.as_ref())
+    .validate_both(schema)
+    .validate_both(definitions)
+    .validate_both(server)?;
```

Comment:
Why not use `flatten` in https://crates.io/crates/tuples

```suggestion
  let (_, schema, definitions, server) = valid_base_url(upstream.base_url.as_ref())
    .validate_both(schema)
    .validate_both(definitions)
    .validate_both(server).map(|a| a.flatten())?;
```